lib/commands/exec.js

```diff
@@ -38,6 +47,18 @@ var command = {
         config.logger.log("Using network '" + config.network + "'." + OS.EOL);
       }
 
+      // `--compile`
+      if (options.c || options.compile){
+        return Contracts.compile(config, function(err){
+          if(err) return done(err);
+
+          Require.exec(config.with({
+            file: file
+          }), done);
+        });
+      };
```

Comment:
This cause `Require.exec` to run twice - maybe we should pull that out so we don't have to maintain that block in two places.